### ISSUE TITLE
Method not found in class

### ISSUE THREAD

**Issue by ethanf22** [barryvdh/laravel-ide-helper#321]
I've run the following three commands:

```
php artisan clear-compiled
php artisan ide-helper:generate
php artisan optimize
```

and I'm still showing a number of functions with errors. 

Many of them say Method not found in class. I also have some that say method signature uses 0 parameters. These functions tend to use `func_get_args()`. 

Some examples are 
`Header::with("campaign")->groupBy("campaign_id") //Method not found on groupBy` 
`JavaScript::put([]); //Method signature uses 0 params`
`Product::where() ;//method not found`
`Product::find(); //method not found`
`$model_var->save(); //method not found, seems to only be an error when ::find is used to get the model`

I've tried adjusting my `use` statements to use the Facade instead of the full namespace and that helped with a few of the errors. The remaining errrors seem to be from Models that don't like the Eloquent functions.

I could turn off the error notification for these but it still won't let me do the autocompletion for the method not found errors. 

I had the Laravel plugin installed but I tried removing that and it had no effect. I also tried setting up my directories and making `/app` and `/vendor` source folders and `/public` a resource root but that had no effect either.

Am I missing something here?

**Comment by truetamtam** [comment]
Same for me.
Fresh laravel install.

**Comment by barryvdh** [comment]
Can you try with the latest version, it should add a `@mixin` tag, which should add most methods.

**Comment by ethanf22** [comment]
I updated to 2.1.4, deleted the _ide_helper.php file and regenerated it. I invalidated the cache and restarted PhpStorm too.

I just now found this line in the docs: "The generator tries to identify the real class, but if it cannot be found, you can define it in the config file." Is this something I have to do to fix this?

**Comment by ethanf22** [comment]
I discovered from past issues that most of these will be fixed by changing `extends Model` to `extends \Eloquent`. I tried to fix the others by using this:

```
'magic' => array(
	    'Eloquent' => array(
		    'groupBy'   => 'Illuminate\Database\Query\Builder::groupBy'
	    )
),
```
in the config file. However PhpStorm now marks it as @internal.

The other remaining issue is with functions that use `func_get_args()`. Is there a way to indicate for the generator what functions these are?
